clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -5279,6 +5329,13 @@ static void handleIntelFPGARegisterAttr(Sema &S, Decl *D,
   if (checkIntelFPGARegisterAttrCompatibility(S, D, Attr))
     return;
 
+  if (Attr.getAttributeSpellingListIndex() ==
+      IntelFPGARegisterAttr::CXX11_intelfpga_register) {
+    S.Diag(Attr.getLoc(), diag::warn_attribute_spelling_deprecated) << Attr;
+    S.Diag(Attr.getLoc(), diag::note_spelling_suggestion)
+        << "'intel::fpga_register'";
+  }
```

Comment:
It there a way to move this part of code to a function? Because it is repeated many times.

Reply:
@Fznamznon, I thought about to move this in a separate function. The issue is that we are telling users to use different attribute spelling in note instead of the deprecated attribute one , not sure how can we do it. Any thought?

Reply:
I don't know how to get a string for desired spelling from the attribute object, unfortunately (maybe @erichkeane can suggest something?). If there is no way to do it, we could just pass a string with desired spelling as a function parameter.

Reply:
We don't have a good way to do that unfortunately, best we  have is Attr::getSpelling, which gets the current spelling.

Reply:
>@Fznamznon, I thought about to move this in a separate function. The issue is that we are telling users to use different attribute spelling in note instead of the deprecated attribute one , not sure how can we do it. Any thought?

Could either be passed in as a parameter along with the attribute kind and spelling, or we could just have one function that checks ALL of them?

Reply:
Thanks for the suggestion. I have moved the diagnostic in a separate function.